exchanges/dydx/dydx_types.go

```diff
@@ -0,0 +1,903 @@
+		TickSize                         float64   `json:"tickSize,string"`
+		IndexPrice                       float64   `json:"indexPrice,string"`
+		OraclePrice                      string    `json:"oraclePrice"`
+		PriceChange24H                   string    `json:"priceChange24H"`
+		NextFundingRate                  string    `json:"nextFundingRate"`
+		NextFundingAt                    time.Time `json:"nextFundingAt"`
+		MinOrderSize                     string    `json:"minOrderSize"`
+		Type                             string    `json:"type"`
+		InitialMarginFraction            string    `json:"initialMarginFraction"`
+		MaintenanceMarginFraction        string    `json:"maintenanceMarginFraction"`
+		BaselinePositionSize             string    `json:"baselinePositionSize"`
+		IncrementalPositionSize          string    `json:"incrementalPositionSize"`
+		IncrementalInitialMarginFraction string    `json:"incrementalInitialMarginFraction"`
+		Volume24H                        float64   `json:"volume24H,string"`
+		Trades24H                        float64   `json:"trades24H,string"`
+		OpenInterest                     string    `json:"openInterest"`
+		MaxPositionSize                  string    `json:"maxPositionSize"`
+		AssetResolution                  string    `json:"assetResolution"`
+		SyntheticAssetID                 string    `json:"syntheticAssetId"`
+	} `json:"markets"`
+}
+
+// MarketOrderbook represents  bids and asks that are fillable are returned.
+type MarketOrderbook struct {
+	Bids orderbookDatas `json:"bids"`
+	Asks orderbookDatas `json:"asks"`
+}
+
+// OrderbookData represents asks and bids price and size data.
+type OrderbookData struct {
+	Offset string  `json:"offset"`
+	Price  float64 `json:"price,string"`
+	Size   float64 `json:"size,string"`
+}
+
+type orderbookDatas []OrderbookData
+
+func (a orderbookDatas) generateOrderbookItem() []orderbook.Item {
+	books := make([]orderbook.Item, len(a))
+	for x := range a {
+		books[x] = orderbook.Item{
+			Price:  a[x].Price,
+			Amount: a[x].Size,
+		}
+	}
+	return books
+}
+
+// MarketTrades represents trade information for specific market(instrument).
+type MarketTrades struct {
+	Trades []MarketTrade `json:"trades"`
+}
+
+// MarketTrade represents a market trade item.
+type MarketTrade struct {
+	Side        string    `json:"side"`
+	Size        float64   `json:"size,string"`
+	Price       float64   `json:"price,string"`
+	CreatedAt   time.Time `json:"createdAt"`
+	Liquidation bool      `json:"liquidation"`
+}
+
+// WithdrawalLiquidityResponse represents accounts that have available funds for fast withdrawals.
+type WithdrawalLiquidityResponse struct {
+	LiquidityProviders map[string]LiquidityProvider `json:"liquidityProviders"`
+}
+
+// LiquidityProvider represents a liquidation provider item data
+type LiquidityProvider struct {
+	AvailableFunds string      `json:"availableFunds"`
+	StarkKey       string      `json:"starkKey"`
+	Quote          interface{} `json:"quote"`
```

Comment:
Can you please change available funds to float64 and go over all types again to make sure they represent float64. Also quote should be a more meaningful type please change across PR.